app/src/main/java/com/leoart/koreanphrasebook/data/repository/DictionaryRepository.kt

```diff
@@ -50,6 +50,28 @@ class DictionaryRepository(val context: Context) {
                 }
     }
 
+    override fun isEmpty(): Single<Boolean> {
+        return AppDataBase.getInstance(context).dictionaryDao().count().flatMap {
+            Single.just(it == 0)
+        }
+    }
+
+    private fun clearDB(){
+        AppDataBase.getInstance(context).dictionaryDao().deleteAll()
+    }
+
+    override fun refreshData(): Completable {
+        return Completable.create { emitter ->
+            DictionaryRequest().getDictionary()
```

Comment:
please create kotlin extension `.flatmapCompletable { Completable.complete() }` and use it everywhere instead wrapping observable into completable